nsapass

```diff
@@ -379,7 +402,8 @@ p_mod.add_argument('index', metavar='INDEX', type=str,
                     help="password index")
 p_mod.add_argument('-n', metavar='NEWNAME', type=str,
                     help="a new username")
-p_mod.add_argument('-p', metavar='NEWPASS', type=str,
+p_mod.add_argument('-p', metavar='NEWPASS', type=str, nargs='?',
+                    default=False,
```

Comment:
thanks to this trick, `-P` is not needed:

- if `-p` is not used, `p_mod.p`'s value is `False`.
- if used, but without an argument `p_mod.p`'s value is `None`.
- if used and with an argument, `p_mod.p` has the argument as a string.

this way, code above doesn't need to enforce mutual exclusivity between `-p` and `-P`. 

this is now also consistent with `nsa_add` which prompts if password's value is not supplied.

Reply:
I don't think that works well from a user perspective, because the handling of optional arguments leads to surprises:
```
$ ./nsapass mod -p name 0
usage: nsapass mod [-h] [-n NEWNAME] [-p [NEWPASS]] [-c NEWNOTE] NAME INDEX
nsapass mod: error: the following arguments are required: INDEX
$ ./nsapass mod name 0 -p
[...p=None...]
$ ./nsapass mod -p -c note name 0     # confusingly, this treats the -c as the next flag
[...p=None...]
$ ./nsapass mod -p-- -c note name 0   # I found this by accident, why does it work?
[...p=None...]
$ ./nsapass mod -p -- -c note name 0  # the space matters
usage: nsapass [-h] [-d] [-z] {get,add,gen,del,undel,mod,list,changepass} ...
nsapass: error: unrecognized arguments: name 0
```
A possible convention that avoids this trouble would be to *require* equality signs, so `-p password` would not work, `-p=password` would provide the argument, and `-p` would use the default. But [argparse](https://docs.python.org/3.8/library/argparse.html) doesn't seem to support that.

Reply:
oops.  will put `-P` back.  

i had a feeling that life can't be this easy 😄 i even wrote it (but then deleted it to reduce my chitchat).  thanks a lot for your time and sorry for the needless hassle i brought.

Reply:
But looking at the documentation, have you considered using `ArgumentParser.add_mutually_exclusive_group`? That sounds like it should support `[-p [NEWPASS] | -P]` kind of things, which is what we want.

Reply:
ya. just did it in the latest force-push.  (racing condition).

Reply:
Yes, putting `-P` first is an excellent idea, I like the usage of `mod` now.

There's an issue with `add`, namely it's not possible to specify a note without also specifying a password.
```
usage: nsapass add [-h] NAME [PASS] [NOTE]
```
Perhaps
```
usage: naspass add [-h] [-p PASS] NAME [NOTE]
```
would work better?

Reply:
oo.  ya.  thanks a lot for bringing this up.

no worries, i totally agree.  i'm optimistic we can nail it.

some days ago, i've also opened an issue to discuss other ui issues concerning `nsapass add` for next major version (version 1).  since we use semantic versioning, version 1 just means that we're free to add improvements even those that break backwards compatibility.

for `nsa_mod`:

## option 1
how about this:

`nsapass add [-h] [-p PASS] [-u NAME] [-c NOTE]`

then, the user gets prompted for all of which is not specified.

## option 2
how about we also merge `nsapass gen` into `nsapass add`?  something like this:

`nsapass add [-h] [-p PASS] [-l LEN -s SET] [-u NAME] [-c NOTE]`

and prompt the user for things not specified in arguments.

because from a usability prespective, i don't think it is convenient to require a user to plan ahead whether the password is manual or generated.  i think users usually are goal oriented (e.g. wanting to add a user), and then they'd like to resolve details later (they may even change their mind midways during the process).  so i think most convenient ui is to rely on `nsapass add` then somehow let the user end up with what he wants with minimum steps.

but currently, a user who starts `nsapass add ...` may end up realizing that he actually needed `nsapass gen ...` which is inconvenient as he will need to rewrite all of the commands.

Reply:
There's value in being consistent, and it would be nice to keep the simple use of `get`. So with that in mind, the `NAME INDEX` part should be consistent across all commands that use it. It's sane to only make `INDEX` optional for the non-destructive `get` command, though maybe an exception could be made for names with only a single used index (that must be 0). So with all that in mind I would not introduce a `-u` option. The rest makes sense to me...
```
usage: nsapass get [-h] NAME [INDEX]
usage: nsapass add [-h] [-p PASS] [-c NOTE] NAME
usage: nsapass gen [-h] [-l LEN] [-s SET] [-c NOTE] NAME
usage: nsapass del [-h] NAME INDEX
usage: nsapass undel [-h] NAME INDEX
usage: nsapass mod [-h] [-n NEWNAME] [-P | -p NEWPASS] [-c NEWNOTE] NAME INDEX
usage: nsapass list [-h] [-a]
usage: nsapass changepass [-h]
```
This assumes that there are some sensible defaults for `-l` and `-s`.

Merging `add` and `gen` is possible but may it overloads the `add` command too much?
```
usage: nsapass add [-h] [-c NOTE] [-P | -p PASS | [-l LEN] [-s SET]] NAME
```